### PR TITLE
Advance HW7 - add 'su' command

### LINKED ISSUE #255: create a reference for "su"
create a reference for "su" command.

### PR DESCRIPTION
**Pull request by zhuoranh**
# Advanced HW 7 submission

Closes #255 

Name: Zhuoran Han
Uniqname:zhuoranh

### CODE REVIEW

File: _commands/basics/su.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+su
+---
+
+`su` (substitute user) is a command to change a login session's owner
+without the owner having to first log out of the session. (as specified 
+in the [su Command](https://www.techonthenet.com/linux/commands/su.php)). In
+other words, it is called the switch user command
+
+__Note:__ It is most commonly employed to change the ownership from
+an ordinary user to the administrative user. Therefore it is oftern
```
Comment (mterwill) [inline _commands/basics/su.md]: s/oftern/often

File: _commands/basics/su.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+~~~
+
+<!--more-->
+
+### Example
+
+~~~ bash
+$ su root
+Password:
+$ sudo alex
```
Comment (mterwill) [inline _commands/basics/su.md]: `sudo alex` would run the command `alex` as root. Is that what you are trying to do here?

File: _commands/basics/su.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+Password:
+~~~
+
+By default, the root account is disabled and doesn't have any password. To create
```
Comment (mterwill) [inline _commands/basics/su.md]: Merp. This is generally considered bad practice – it's best to leave the root account disabled and escalate only when necessary. Let's nix this example.

File: _commands/basics/su.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+~~~ bash
+$ su -c 'ls /home/alex'
+~~~
+
+`su -` will change the current directory and environment to what would be expected if the new user
+had logged on to a new session. The following example will lead to the home directory of a user named
+alex.
+
+~~~ bash
+$ su - alex
```
Comment (mterwill) [inline _commands/basics/su.md]: Make sure you note here that this prompts for alex's password. If you are a superuser on the box and want to masquerade as alex you can definitely:

```
$ sudo su - alex
```

File: _commands/basics/su.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+other words, it is called the switch user command
+
+__Note:__ It is most commonly employed to change the ownership from
+an ordinary user to the administrative user. Therefore it is oftern
+refered as the superuser command.
+
+
+
+~~~ bash
+$ su root
```
Comment (mterwill) [inline _commands/basics/su.md]: This won't work unless root has a password set. You'd have to `sudo su root`, which is exactly the same as `sudo su`, which I use most often.

**Comment by zhuoranh** [comment]
Revised based on your suggestion. @mterwill

File: _commands/basics/su.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+[sudo] password for alex:
+~~~
+
+Here it asks for alex's password not root password.
+
+If you are a superuser on the box and would like to masquerade as user alex,
+you can run:
+
+~~~ bash
+$ sudo su -alex
```
Comment (mterwill) [inline _commands/basics/su.md]: Looks like you're missing a space here. This would be parsed as `su` with arguments `alex`.

Comment (mterwill) [inline _commands/basics/su.md]: It also looks like you left the example beginning on like 62, which is now a duplicate.

**Comment by zhuoranh** [comment]
@mterwill Thanks!

**Comment by mterwill** [comment]
Thanks for fixing everything up! Merging + giving you credit.

**Pull request merged**